src/extension/csw/api/src/main/java/org/geoserver/csw/records/GenericRecordBuilder.java

```diff
@@ -55,33 +56,25 @@
         AttributeDescriptor descriptor = null;
 
         public abstract TreeNode clone();
-    }
-
-    protected abstract static class TreeLeaf extends TreeNode {
-        public Map<Object, Object> userData;
-    }
-
-    protected static class ComplexTreeLeaf extends TreeLeaf {
-        public Map<String, Object> value = new HashMap<String, Object>();
 
-        @Override
-        public ComplexTreeLeaf clone() {
-            ComplexTreeLeaf leaf = new ComplexTreeLeaf();
-            leaf.value.putAll(value);
-            leaf.userData = userData;
-            leaf.descriptor = descriptor;
-            return leaf;
+        public boolean cleanUp() {
+            // TODO Auto-generated method stub
```

Comment:
?

Reply:
todo needs to be removed there indeed and better comment added